sys/include/net/gnrc/ipv6.h

```diff
@@ -11,10 +11,63 @@
  * @ingroup     net_gnrc
  * @brief       GNRC's IPv6 implementation
  *
- * The IPv6 control thread understands messages of type
- *
- *  * @ref GNRC_NETAPI_MSG_TYPE_RCV, and
- *  * @ref GNRC_NETAPI_MSG_TYPE_SND,
+ * This module is for usage with the @ref net_gnrc_netapi
+ *
+ * # NETAPI command documentation
+ *
+ * ## `GNRC_NETAPI_MSG_TYPE_RCV`
+ *
+ * @ref GNRC_NETAPI_MSG_TYPE_RCV expects a @ref net_gnrc_pkt (referred to as
+ * "packet" in the following) in receive order (payload first, headers in
+ * decreasing order by layer). It should at least contain a link-layer payload.
+ * If it contains a @ref net_gnrc_netif_hdr it will be taken into account.
+ *
+ * If the link-layer payload is not an IPv6 packet (starting with a 0x6
+ * half-byte) or the module is not able to parse it correctly it will be
+ * dropped. If the packet can't be forwarded (either to another node or an
+ * application) it will be dropped. Otherwise, there are two scenarios of
+ * success:
+ *
+ * 1. If the destination address is an address on this host, and a suitable
+ *    application is registered to the next header field of the IPv6 header,
+ *    the next header field of one of the IPv6 extension headers within the
+ *    packet or @ref gnrc_nettype_from_protnum() resolves one of the previously
+ *    mentioned next header fields to a @ref net_gnrc_nettype, the packet will
+ *    be forwarded to that application, with the IPv6 header and every IPv6
+ *    extension header between the IPv6 header and the payload marked as
+ *    separate @ref gnrc_pktsnip_t (i.e. the IPv6 payload will be the first
+ *    snip due to receive order). Note, that IPv6-internal headers (such as
```

Comment:
This long sentence is hard to parse. Can this be simplified to become one step one sentence?

Reply:
Broke it up to be easier to understand